ir/type.cpp

```diff
@@ -482,57 +491,95 @@ void ArrayType::print(ostream &os) const {
   os << "TODO";
 }
 
+static std::vector<std::unique_ptr<IntType>> child_types;
+static unsigned child_ty_num;
+
+ // TODO: Support symbolic vector on floating point types
+VectorType::VectorType(std::string &&name) : Type(std::move(name)) {
+  elementTy = child_types.emplace_back(
+    std::make_unique<IntType>("child_" + std::to_string(child_ty_num++))).get();
```

Comment:
it should receive the element type as input. Even if it's a symbolic type, for example.

Reply:
Dear @nunoplopes , I don't catch your comment, what do you mean by "receive element types as input", does it mean to drop this constructor?  We do have another constructor with takes element type as its argument. 

Also allowing vector of symbolic types may leads to infinite recursions, like vector of vector of endless of vectors.... I am thinking of creating a new ScalarType class. IntType, FloatType and Ptr types are inherited from the new class and VectorType can only produce vector on scalars.